server/app/services/doctor_service.py

```diff
@@ -75,3 +79,58 @@ async def set_location_for_doctor(self, doctor_email: str, address: Location):
             return 200, "Address updated successfully"
         except Exception as e:
             return 500, e
+
+
+    async def get_doctor_by_email(self, email: str):
+        """
+        Fetch a doctor's details by email.
+
+        Args:
+            email (str): The email of the doctor.
+
+        Returns:
+            dict: A dictionary containing the doctor's details or None if not found.
+        """
+        try:
+            doctor = await self.doctor_collection.find_one({"email": {"$eq": email}})
+            if doctor:
+                return {
+                    "name": doctor["name"],
+                    "email": doctor["email"],
+                    "specialisation": doctor.get("specialisation"),
+                    "pincode": doctor.get("pincode")
+                }
+            return None
+        except Exception as e:
+            print(f"An error occurred while fetching doctor by email: {e}")
+            return None
+    
+    async def save_availability(self, availability: DoctorSchedule):
+        """
+        Save a doctor's availability to the database.
+
+        Args:
+            availability (DoctorSchedule): The availability object to save.
+
+        Returns:
+            The saved availability object or an error message.
+        """
+        try:
+            # Check if availability already exists for the doctor
+            existing_availability = await self.availability_collection.find_one({"doctor_email": availability.doctor_email})
+            print(f"Existing availability for {availability.doctor_email}: {existing_availability}")  # Debugging line
+
+            if existing_availability:
+                # Update the existing availability 
+                await self.availability_collection.update_one(
+                    {"doctor_email": availability.doctor_email},
+                    {"$set": availability.dict()}
+                )
+                return availability  # Return updated availability
+            else:
+                # Logic to save new availability in the database
+                await self.availability_collection.insert_one(availability.dict())
```

Comment:
change availability to schedule.

Reply:
its saved as doctorschedule in database. this is just the variable getting used here.